packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -473,11 +503,34 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
 
     const renderedColumns = visibleColumns.slice(firstColumnToRender, lastColumnToRender);
 
+    // If the selected column is not within the current range of columns being displayed,
```

Comment:
There's a focus jump when clicking on the cell after the focused cell was moved out of view: https://deploy-preview-7357--material-ui-x.netlify.app/x/react-data-grid/virtualization/

https://user-images.githubusercontent.com/13808724/212136876-48905cf4-bc80-4e00-a41d-d63fcfd690e7.mov

Reply:
The focused cell was causing the render zone to shift to the right, resulting in the appearance of a jumping focus. To resolve this, I have set the cell width and opacity to zero when using column virtualization.